### ISSUE TITLE
CMAKE_INSTALL_PREFIX make configuration to fail

### ISSUE THREAD

**Issue by chanfr** [aws/aws-sdk-cpp#1156]
### What platform/OS are you using?
Both Linux/Windows

### Which version of the SDK?
1.7.110 and master

### What compiler are you using? what version?
gcc and vc14

### What are your CMake arguments?
cmake .. -DCMAKE_INSTALL_PREFIX=../install

### Can you provide a TRACE level log? (sanitize any sensitive information)

CMake Error at aws-cpp-sdk-core/CMakeLists.txt:502 (find_package):
  By not providing "Findaws-c-event-stream.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "aws-c-event-stream", but CMake did not find one.

  Could not find a package configuration file provided by
  "aws-c-event-stream" with any of the following names:

    aws-c-event-streamConfig.cmake
    aws-c-event-stream-config.cmake

  Add the installation prefix of "aws-c-event-stream" to CMAKE_PREFIX_PATH or
  set "aws-c-event-stream_DIR" to a directory containing one of the above
  files.  If "aws-c-event-stream" provides a separate development package or
  SDK, be sure it has been installed.

-- Configuring incomplete, errors occurred!
See also "/home/frivas/.repo/3rd-party/aws/1.7.110/code/CMakeFiles/CMakeOutput.log".
See also "/home/frivas/.repo/3rd-party/aws/1.7.110/code/CMakeFiles/CMakeError.log".

For both os Im getting the same error. 

Thanks in advance.

**Comment by marcomagdy** [comment]
The SDK has third-party dependencies that are built under the `.deps` directory.
- If you specify an install prefix, the third-party libs will be installed under that prefix.
- If you do not specify an install prefix, they will be installed under the `.deps` directory.
- If you do not specify an install prefix initially, and then later specify one, that will only affect the SDK and the third-party libs will remain installed under their initial prefix.

Bottom line: if you want to _change_ the install prefix, you have to remove the (hidden) `.deps` directory and run cmake again.

**Comment by chanfr** [comment]
Hi,
thanks for the answer. But it does not work by default. I mean... in a clean clone

I just execute:
cmake . -DCMAKE_INSTALL_PREFIX=custom_install -DBUILD_ONLY="s3;transfer"

and the during the cmake configuration it fails. It is the expected behavior?

See log attached
[log.log](https://github.com/aws/aws-sdk-cpp/files/3276298/log.log)

regards,

**Comment by singku** [comment]
Looking at your log
3rd-party dependencies have been installed to `/mnt/8611eb5f-d411-4940-b345-f08d7f966389/fake_repo/3rd-party/aws/raw/.deps/custom_install/`

You need to remove the build directory and do a clean build

**Comment by ingomueller-net** [comment]
>     * If you specify an install prefix, the third-party libs will be installed under that prefix.

This is not actually true (currently, i.e., version 1.7.138): the third-party libraries will *not* be installed to the prefix (they will not even be built) -- they are assumed to be installed there already. I think that this is misusing `CMAKE_INSTALL_PREFIX`; it should actually do what you said.
